libsrc/_DEVELOPMENT/temp/sp1/zx/sprites/asm_sp1_AddColSpr.asm

```diff
@@ -33,9 +33,19 @@ csalloc:
    push bc
    ld hl,24                   ; sizeof(struct sp1_cs)
    push hl
+
+; classic malloc exits with carry set if success, reset on failure
+; newlib asm_malloc exits with carry reset if success, set on failure
```

Comment:
I don't think this is correct: u_malloc() is a C function so "can't" return with a carry flag condition.

Reply:
I understand that, but indeed the asm code in the SP1 functions check for this C or NC flag, and they are inverted in the asm heap alloc routines... The asm code is _using_ the Carry flag to check for errors in both places...

Anyway, I just put inside the IF-ENDIF the code that was different between the Classic and the Newlib code, so I guess this is harmless re: the PR content.

I think I have an explanation: maybe this is a bug on the original Classic-SP1 sources: supposedly you could use your own memory allocation routines by setting `u_malloc` and `u_free` pointers, but indeed the SP1 code was using the Carry check for errors, and not the return value. And because everyone just set `u_malloc = malloc` and `u_free = free`, and everything worked, nobody noticed.

So you can set your own routines...as long as you end up quickly calling the internal `malloc`, `HeapAlloc`, etc. ones...which set/reset the Carry flag for error reporting (which is what SP1 expects).

Reply:
Yes, that would be that it worked through pure luck - the examples implement u_malloc which just calls malloc directly so carry would be preserved. 

The luck seems to be holding....

Reply:
...and may be this is the reason that Newlib-SP1 sources just call `asm_malloc` instead of using the alternate pointers.